QSB/Instruments/InstrumentsManager.cs

```diff
@@ -10,54 +10,56 @@ namespace QSB.Instruments
 		{
-			rootObj = root;
+			_rootObj = root;
 			gameObject.AddComponent<CameraManager>();
 
-			QSBInputManager.ChertTaunt += () => StartInstrument(AnimationType.Chert);
-			QSBInputManager.EskerTaunt += () => StartInstrument(AnimationType.Esker);
-			QSBInputManager.FeldsparTaunt += () => StartInstrument(AnimationType.Feldspar);
-			QSBInputManager.GabbroTaunt += () => StartInstrument(AnimationType.Gabbro);
-			QSBInputManager.RiebeckTaunt += () => StartInstrument(AnimationType.Riebeck);
-			QSBInputManager.ExitTaunt += () => ReturnToPlayer();
+			QSBInputManager.ChertTaunt += OnChertTaunt;
+			QSBInputManager.EskerTaunt += OnEskerTaunt;
+			QSBInputManager.FeldsparTaunt += OnFeldsparTaunt;
+			QSBInputManager.GabbroTaunt += OnGabbroTaunt;
+			QSBInputManager.RiebeckTaunt += OnRiebeckTaunt;
+			QSBInputManager.ExitTaunt += ReturnToPlayer;
 
-			QSB.Helper.Events.Unity.RunWhen(() => Locator.GetPlayerBody() != null, SetupInstruments);
-
-			QSBPlayerManager.PlayerSyncObjects.Add(this);
+			QSBCore.Helper.Events.Unity.RunWhen(() => Locator.GetPlayerBody() != null, SetupInstruments);
 		}
 
 		public void InitRemote(Transform root)
 		{
-			rootObj = root;
-			QSB.Helper.Events.Unity.RunWhen(() => Locator.GetPlayerBody() != null, SetupInstruments);
-
-			QSBPlayerManager.PlayerSyncObjects.Add(this);
+			_rootObj = root;
+			QSBCore.Helper.Events.Unity.RunWhen(() => Locator.GetPlayerBody() != null, SetupInstruments);
 		}
 
-		private void OnDestroy()
+		protected override void OnDestroy()
 		{
+			base.OnDestroy();
 			if (!IsLocalPlayer)
 			{
 				return;
 			}
-			DebugLog.DebugWrite($"OnDestroy {PlayerId}");
-			QSBInputManager.ChertTaunt -= () => StartInstrument(AnimationType.Chert);
-			QSBInputManager.EskerTaunt -= () => StartInstrument(AnimationType.Esker);
-			QSBInputManager.FeldsparTaunt -= () => StartInstrument(AnimationType.Feldspar);
-			QSBInputManager.GabbroTaunt -= () => StartInstrument(AnimationType.Gabbro);
-			QSBInputManager.RiebeckTaunt -= () => StartInstrument(AnimationType.Riebeck);
-			QSBInputManager.ExitTaunt -= () => ReturnToPlayer();
+			QSBInputManager.ChertTaunt -= OnChertTaunt;
+			QSBInputManager.EskerTaunt -= OnEskerTaunt;
+			QSBInputManager.FeldsparTaunt -= OnFeldsparTaunt;
+			QSBInputManager.GabbroTaunt -= OnGabbroTaunt;
+			QSBInputManager.RiebeckTaunt -= OnRiebeckTaunt;
+			QSBInputManager.ExitTaunt -= ReturnToPlayer;
 		}
 
+		private void OnChertTaunt() => StartInstrument(AnimationType.Chert);
+		private void OnEskerTaunt() => StartInstrument(AnimationType.Esker);
+		private void OnFeldsparTaunt() => StartInstrument(AnimationType.Feldspar);
+		private void OnGabbroTaunt() => StartInstrument(AnimationType.Gabbro);
+		private void OnRiebeckTaunt() => StartInstrument(AnimationType.Riebeck);
+
 		private void SetupInstruments()
 		{
-			var bundle = QSB.InstrumentAssetBundle;
-			ChertDrum = MakeChertDrum(bundle);
+			var bundle = QSBCore.InstrumentAssetBundle;
+			_chertDrum = MakeChertDrum(bundle);
 		}
```

Comment:
I see you like expression methods, you could easily turn this into one if you like, by first making it a one-liner. Your call ;)

Reply:
would just undo that when I add more instruments